slo_generator/report.py

```diff
@@ -111,20 +126,34 @@ def __init__(self,
         self.valid = True
         self.errors = []
 
+        # Get Assumption of Empty SLI value in slo-config
+        #assumption_empty_sli = self.get_assumption_empty_sli(config)
+        assumption_empty_sli = 100
+
         # Get backend results
-        data = self.run_backend(config, backend, client=client, delete=delete)
-        if not self._validate(data):
+        # Patch globo-0.3 for three retries
+        RETRIES=3
+        while RETRIES > 0:
+            data = self.run_backend(config, backend, client=client, delete=delete)
+            LOGGER.debug(f'patch globo-0.3 | After get data in backend({str(data)}).')
+            if not self._validate(data, assumption_empty_sli):
+                RETRIES -= 1
+                rWAIT = random.randint(2, 5)
+                LOGGER.debug(f'patch globo-0.3 | Waiting {rWAIT} before next try. ({RETRIES} retries remaining)')
+                time.sleep(rWAIT)
+            else:
+                RETRIES = 0
+        if not self._validate(data, assumption_empty_sli):
+            LOGGER.error('patch globo-0.3 | an error occurs in last validation of retrieved data from backend...')
             self.valid = False
             return
-
         # Build SLO report
-        self.build(step, data)
-
+        self.build(step, data, assumption_empty_sli)
         # Post validation
         if not self._post_validate():
             self.valid = False
 
-    def build(self, step, data):
+    def build(self, step, data, assumption_empty_sli=-1):
```

Comment:
same here, I don't think we should pass `assumption_empty_sli` to the build method.
imho the way should be a user configurable global variable to specify the SLI value when no events are found.
Please keep in mind that SLOs with no events cannot be relied on, read https://sre.google/workbook/alerting-on-slos/#low-traffic-services-and-error-budget-alerting for more info on why.